clarity/src/vm/functions/database.rs

```diff
@@ -769,6 +770,29 @@ pub fn special_get_block_info(
         _ => return Ok(Value::none()),
     };
 
+    // interpret height as a tenure height IFF
+    // * clarity version is less than Clarity3
+    // * the evaluated epoch is geq 3.0
+    // * we are not on (classic) primary testnet
+    let interpret_height_as_tenure_height = env.contract_context.get_clarity_version()
+        < &ClarityVersion::Clarity3
+        && env.global_context.epoch_id >= StacksEpochId::Epoch30
+        && env.global_context.chain_id != CHAIN_ID_TESTNET;
```

Comment:
So this applies in Nakamoto testnets and mainnet, but not primary testnet?  Can we open an issue to remove that last gating once we have a way to patch the primary testnet chainstate in-situ?

Reply:
https://github.com/stacks-network/stacks-core/issues/5326